### DISCUSSION TITLE
How can we see from the Kafka UI the messages that are processed and consumed by the consumer?

### DISCUSSION THREAD

**Discussion by Thilaknath** [provectus/kafka-ui#3657] [Q&A] [answered]
We see different views on Topics, Brokers, Consumers. But we would like to know at a given point in time how much messages are currently present in the queue waiting to be processed and how much are already consumed?

Is there a way we can achieve the above?

**Comment by Haarolean** [comment, marked as answer]
Hi.

Regarding the "how many messages are present in the queue":
It's called consumer lag, in our app could be found in consumers' view as "Messages Behind" column.

Regarding "how many are already consumed":
That's nearly impossible. You can't know which consumer consumed which message which, as well, is no longer present in the topic btw.

**Reply by Thilaknath** [reply]
Thank you for the clarification
